nativelib/src/main/resources/time.c

```diff
@@ -30,3 +30,28 @@ long long scalanative_nano_time() {
 
   return nano_time;
 }
+
+long long scalanative_current_time_millis() {
+	long long current_time_millis;
+
+#define NANOS_PER_MILLI 1000000
+#define MILLIS_PER_SEC 1000
+#define MICROS_PER_MILLI 1000
+
+#ifdef __MACH__
+	struct timeval  tv;
+	gettimeofday(&tv, NULL);
```

Comment:
The `clock_gettime` is the POSIX version so my thinking is we should try and be as POSIX compliant as possible to be the most portable - supposedly it is available on Mac OS after a certain version but I had no luck. http://www.unix.com/man-page/POSIX/3posix/clock_gettime/